### ISSUE TITLE
[install] Remove Xorg in favor of Xwayland

### ISSUE THREAD

**Issue by LukeShortCloud** [winesapOS/winesapOS#891]
Look into if it is possible for us to replace Xorg entirely with Xwayland. This provides lots of features such as better battery life, HDR, VRR, etc.

**Comment by GuestSneezeOSDev** [comment]
Uncomment `#WaylandEnable=false` in /etc/gdm/custom.conf line

**Comment by LukeShortCloud** [comment]
For the display manager (DM), we previously used LightDM but recently switched to the Simple Desktop Display Manager (SDDM) because (1) it's managed by the KDE team and (2) it supports Wayland. We don't use the GNOME Display Manager (GDM) because, although it technically can work with other desktop environments and sessions, it's still very GNOME specific.

**Comment by GuestSneezeOSDev** [comment]
Replacing Xorg with Xwayland may not support `i3` and `sway`

**Comment by LukeShortCloud** [comment]
`i3` won't work,  you're right, but `sway` is supposed to be the drop-in Wayland replacement.

**Comment by GuestSneezeOSDev** [comment]
if the DE is i3 install i3 and remove wayland if the DE is Sway install wayland and remove xorg

**Comment by LukeShortCloud** [comment]
That's a good idea. Our first-time setup script can handle that. I just need to figure out how to launch that by default with a window tiling manager. From my testing, it only seems to run in desktop environments.

**Comment by GuestSneezeOSDev** [comment]
@LukeShortCloud have you had any luck?

**Comment by LukeShortCloud** [comment]
The more I look into this, the more I think it's not possible in Arch Linux. Fedora seems to be the only major distro that has thoroughly gone through their packages and separated out Xorg and Wayland dependencies.

For example, the [steam](https://gitlab.archlinux.org/archlinux/packaging/packages/steam/-/blob/main/PKGBUILD?ref_type=heads) package in Arch Linux has a required dependency on `xorg-xrandr`. Wayland does not have a drop-in replacement for that.

The easiest thing we can do from a distribution perspective is to not ship desktop environments that use Xorg. However, there will still be various Xorg dependencies installed on the system. Any Wayland desktop environment should default to using Xwayland which means those Xorg libraries will just be wasted space.

**Comment by LukeShortCloud** [comment]
I've been testing Plasma 6.1 and 6.2 on Mesa 24.1 with NVIDIA hardware and it's very buggy. Lots of crashes. We can wait for NVK to mature before dropping the Xorg sessions.

**Comment by LukeShortCloud** [comment]
I also tested Mesa 24.2.4 and Mesa 24.3.0-git (latest as of today). The latest version finally works very well with NVIDIA. That is slated to come out in about 1 month. When we have the stable release of Mesa 24.3.0, we can consider dropping Xorg sessions.

**Comment by LukeShortCloud** [comment]
Our first-time setup has some assumptions about Xorg. For example, it uses `xrandr` to handle screen rotation but that tool does not work on Wayland.

**Comment by LukeShortCloud** [comment]
Checklist for removing Xorg support:
- [ ] Do not install `xcb*` and `xorg*` packages
- [ ] Remove i3 support in favor of Sway
- [ ] Use an alternative to `xrandr` in the first-time setup

**Comment by GuestSneezeOSDev** [comment]
> Checklist for removing Xorg support:
> 
> * [x]  Mesa 24.3.0 for better NVK Wayland support
> * [ ]  Do not install `xcb*` and `xorg*` packages
> * [ ]  Remove i3 support in favor of Sway
> * [ ]  Use an alternative to `xrandr` in the first-time setup
>   
>   * Hints: https://bbs.archlinux.org/viewtopic.php?id=296888

Thanks, I'll try to resolve this issue

**Comment by LukeShortCloud** [comment]
Sounds good, thanks in advance!

**Comment by GuestSneezeOSDev** [comment]
> Sounds good, thanks in advance!

Your welcome

**Comment by GuestSneezeOSDev** [comment]
https://github.com/GuestSneezeOSDev/winesapOS/blob/main/scripts/winesapos-install.sh#L765 to https://github.com/GuestSneezeOSDev/winesapOS/blob/main/scripts/winesapos-install.sh#L768 are the lines which install xorg, so we could try to find some packages to replace them

**Comment by GuestSneezeOSDev** [comment]
https://github.com/GuestSneezeOSDev/winesapOS/commit/ca175499db1fb67b5c1f318351e0ecac13e87d44, and also Merry Christmas

**Comment by LukeShortCloud** [comment]
Thank you for helping with all of this (including the Hyprland support)! I also like how you refactored it so that Xorg can still be installed but Wayland is default.

**Comment by GuestSneezeOSDev** [comment]
> Thank you for helping with all of this (including the Hyprland support)! I also like how you refactored it so that Xorg can still be installed but Wayland is default.

Your welcome, lad

**Comment by LukeShortCloud** [comment]
The last thing we need is to update the first-time setup script to use Wayland programs instead of Xorg. I'll work on that.

**Comment by GuestSneezeOSDev** [comment]
Steam requires xorg-xandr

**Comment by LukeShortCloud** [comment]
You're right, I forgot about that. It's a hardcoded dependency of Steam. Funnily enough, `xorg-xrandr` actually does not work at all with Wayland. Steam still works just fine.

At the very least, we have minimized the Xorg dependencies as much as we can.

The next step would also be to ensure that there are no Xorg sessions for Plasma. Today, users have the option of using "Plasma" (the Wayland session) or "Plasma (X11)" (the Xorg session) on the login  page. I'll remove the "Plasma (X11)" option.

**Comment by LukeShortCloud** [comment]
For KDE Plasma on Arch Linux, the `plasma-workspace` package includes sessions for both Xorg and Wayland.

```
plasma-workspace usr/share/wayland-sessions/plasma.desktop
plasma-workspace usr/share/xsessions/plasmax11.desktop
```

For KDE Plasma on Manjaro, the `plasma-workspace` package also includes sessions for both Xorg and Wayland but the name of them is slightly different.

```
plasma-workspace usr/share/wayland-sessions/plasmawayland.desktop
plasma-workspace usr/share/xsessions/plasma.desktop
```

**Comment by LukeShortCloud** [comment]
For simplicity's sake, we could just say that Xorg support is deprecated and will be fully removed in a future release.

The last thing I need to do right now is to replace `xrandr` with a similar Wayland tool.

**Comment by LukeShortCloud** [comment]
For KDE Plasma, `kscreen-doctor` can be used to rotate the screen.

Find the correct display: `kscreen-doctor -o`

Then rotate one of a few different ways:
- Default hardware orientation: `kscreen-doctor "output.<DISPLAY>.rotation.normal`
- Left: `kscreen-doctor "output.<DISPLAY>.rotation.left`
- Right: `kscreen-doctor "output.<DISPLAY>.rotation.right`
- Upside-down: `kscreen-doctor "output.<DISPLAY>.rotation.inverted`

These changes are persistent across reboots.

**Comment by LukeShortCloud** [comment]
Okay, screen rotation has been updated, tested, and it's working! I'll go ahead and close this one out now.
